plugins/woocommerce-blocks/assets/js/data/cart/thunks.ts

```diff
@@ -194,10 +194,12 @@ export const removeCoupon =
  *
  * @param {number} productId    Product ID to add to cart.
  * @param {number} [quantity=1] Number of product ID being added to cart.
+ * @param {array}  [variation] Array of variation attributes for the product.
+ * @param {array}  [additionalData] Array of additional fields for the product.
  * @throws           Will throw an error if there is an API problem.
  */
 export const addItemToCart =
-	( productId: number, quantity = 1 ) =>
+	( productId: number, quantity = 1, variation: Array<any> = [], additionalData: Record<string, any> = {} ) =>
```

Comment:
We have a `lint` rule to not use `any`. I think for the variation you could add a type above this function in this format:
```
type Variation = {
  attribute: string;
  value: string;
}
```
And then use it above like this: `variation: Variation[]`.
For the `additionalData` you can use `unknown`: `additionalData: Record<string, unknown> = {}`